Notepad.Extensions.Logging/WindowEnumerationState.cs

```diff
@@ -47,26 +49,50 @@ public bool ExamineWindow(IntPtr hWnd)
             return true;
         }
 
-        static Regex notepadPlusPlusRegex = new Regex(@"^new \d+ - Notepad\+\+$", RegexOptions.Compiled);
+        static Regex notepadPlusPlusRegex = new Regex(@" - Notepad\+\+$", RegexOptions.Compiled);
 
         bool IsKnownNotepadWindow(string titleText)
         {
-            switch (titleText)
+            if (!string.IsNullOrWhiteSpace(WindowName))
             {
-                case "Untitled - Notepad":
-                    WindowKind = WindowKind.Notepad;
-                    Handle = NativeMethods.FindWindowEx(Handle, IntPtr.Zero, "EDIT", null);
-                    return true;
+                if (WindowName.Equals(titleText, StringComparison.Ordinal))
+                {
+                    WindowKind = notepadPlusPlusRegex.IsMatch(titleText) ? WindowKind.NotepadPlusPlus : WindowKind.Notepad;
+                }
             }
-
-            if (notepadPlusPlusRegex.IsMatch(titleText))
+            else
             {
-                WindowKind = WindowKind.NotepadPlusPlus;
-                Handle = NativeMethods.FindWindowEx(Handle, IntPtr.Zero, "Scintilla", null);
-                return true;
+                switch (titleText)
```

Comment:
Having a `switch` seems odd when there's only one case.. `if/else if/ else if` would probably be better here.

Reply:
Ok, I will change it.